File: shared/src/api/failover.ts
export type FailoverCandidateKind = 'ethernet' | 'wifi' | 'usb';

export type FailoverTrackingState =
  | 'online'
  | 'offline'
  | 'disabled'
  | 'not_installed'
  | 'not_available'
  | 'unknown';

export interface FailoverHealthConfig {
  readonly track_ips: readonly string[];
  readonly reliability: number;
  readonly count: number;
  readonly timeout: number;
  readonly interval: number;
  readonly failure_interval: number;
  readonly recovery_interval: number;
  readonly down: number;
  readonly up: number;
}

export interface FailoverCandidate {
  readonly id: string;
  readonly label: string;
  readonly interface_name: string;
  readonly kind: FailoverCandidateKind;
  readonly available: boolean;
  readonly enabled: boolean;
  readonly priority: number;
  readonly tracking_state: FailoverTrackingState;
  readonly is_up: boolean;
}

export interface FailoverEvent {
  readonly from_interface: string;
  readonly to_interface: string;
  readonly timestamp: number;
  readonly reason: string;
}

export interface FailoverConfig {
  readonly available: boolean;
  readonly service_installed: boolean;
  readonly enabled: boolean;
  readonly active_interface: string;
  readonly candidates: readonly FailoverCandidate[];
  readonly health: FailoverHealthConfig;
  readonly last_failover_event?: FailoverEvent;
}

File: backend/internal/models/failover.go
package models

// FailoverCandidateKind identifies the uplink type shown in the UI.
type FailoverCandidateKind string

const (
	FailoverCandidateKindEthernet FailoverCandidateKind = "ethernet"
	FailoverCandidateKindWiFi     FailoverCandidateKind = "wifi"
	FailoverCandidateKindUSB      FailoverCandidateKind = "usb"
)

// FailoverTrackingState describes the current runtime health summary for a candidate.
type FailoverTrackingState string

const (
	FailoverTrackingStateOnline       FailoverTrackingState = "online"
	FailoverTrackingStateOffline      FailoverTrackingState = "offline"
	FailoverTrackingStateDisabled     FailoverTrackingState = "disabled"
	FailoverTrackingStateNotInstalled FailoverTrackingState = "not_installed"
	FailoverTrackingStateNotAvailable FailoverTrackingState = "not_available"
	FailoverTrackingStateUnknown      FailoverTrackingState = "unknown"
)

// FailoverHealthConfig contains the shared mwan3 tracking parameters used by the app.
type FailoverHealthConfig struct {
	TrackIPs         []string `json:"track_ips"`
	Reliability      int      `json:"reliability"`
	Count            int      `json:"count"`
	Timeout          int      `json:"timeout"`
	Interval         int      `json:"interval"`
	FailureInterval  int      `json:"failure_interval"`
	RecoveryInterval int      `json:"recovery_interval"`
	Down             int      `json:"down"`
	Up               int      `json:"up"`
}

// FailoverCandidate is a single orderable UI row.
type FailoverCandidate struct {
	ID            string                `json:"id"`
	Label         string                `json:"label"`
	InterfaceName string                `json:"interface_name"`
	Kind          FailoverCandidateKind `json:"kind"`
	Available     bool                  `json:"available"`
	Enabled       bool                  `json:"enabled"`
	Priority      int                   `json:"priority"`
	TrackingState FailoverTrackingState `json:"tracking_state"`
	IsUp          bool                  `json:"is_up"`
}

// FailoverEvent describes an observed active-uplink switch.
type FailoverEvent struct {
	FromInterface string `json:"from_interface"`
	ToInterface   string `json:"to_interface"`
	Timestamp     int64  `json:"timestamp"`
	Reason        string `json:"reason"`
}

// FailoverConfig is the top-level GET/PUT contract.
type FailoverConfig struct {
	Available         bool                 `json:"available"`
	ServiceInstalled  bool                 `json:"service_installed"`
	Enabled           bool                 `json:"enabled"`
	ActiveInterface   string               `json:"active_interface"`
	Candidates        []FailoverCandidate  `json:"candidates"`
	Health            FailoverHealthConfig `json:"health"`
	LastFailoverEvent *FailoverEvent       `json:"last_failover_event,omitempty"`
}
